src/services/inlayHints.ts

```diff
@@ -137,10 +137,18 @@ namespace ts.InlayHints {
 
             const typeDisplayString = printTypeInSingleLine(declarationType);
             if (typeDisplayString) {
+                const isVariableNameMatchesType = !preferences.includeInlayVariableTypeHintsWhenTypeMatchesName && variableDeclarationNameMatchesTypeString(decl.name.getText(), typeDisplayString);

```

Comment:
I’m not sure how concerned to be that older clients won’t send this preference and so for them, the default behavior will change. Maybe it’s better to see if the preference `=== false` so that `undefined` is like `true`.